v4l2.go

```diff
@@ -311,6 +336,56 @@ func getFrameSize(fd uintptr, index uint32, code uint32) (frameSize FrameSize, e
 	return
 }
 
+func getFrameInterval(fd uintptr, index uint32, code uint32, width uint32, height uint32) (FrameRate, error) {
+	frmivalEnum := &v4l2_frmivalenum{
+		index:        index,
+		pixel_format: code,
+		width:        width,
+		height:       height,
+	}
+
+	var frameRate FrameRate
+
+	err := ioctl.Ioctl(fd, VIDIOC_ENUM_FRAMEINTERVALS, uintptr(unsafe.Pointer(frmivalEnum)))
+	if err != nil {
+		return frameRate, err
+	}
+
+	switch frmivalEnum._type {
+
+	case V4L2_FRMIVAL_TYPE_DISCRETE:
+		discrete := &v4l2_fract{}
+		err = binary.Read(bytes.NewBuffer(frmivalEnum.union[:]), NativeByteOrder, discrete)
```

Comment:
easier to early return in a short if `if err :=`

Reply:
Changed to shorthand.